### PR TITLE
Make HttpConnector retries configurable and add jitter

### PR DESCRIPTION
**Pull request by exoson**
HttpConnector has some hard coded values for maximum number of attempts and maximum timeout between retries. Make these configurable by command line flags. Also add jitter to those retries.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java [hunk 1]
```diff
@@ -124,6 +124,26 @@ public class RepositoryOptions extends OptionsBase {
       help = "Scale all timeouts related to http downloads by the given factor")
   public double httpTimeoutScaling;
 
+  @Option(
+      name = "http_connector_attempts",
+      defaultValue = "0",
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java:129]: why not just make the default value 8? same for the other flag

File: src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java [hunk 2]
```diff
@@ -124,6 +124,26 @@ public class RepositoryOptions extends OptionsBase {
...
+      documentationCategory = OptionDocumentationCategory.BAZEL_CLIENT_OPTIONS,
+      effectTags = {OptionEffectTag.UNKNOWN},
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java:131]: I'd follow the example of the `http_timeout_scaling` flag and use `BAZEL_INTERNAL_CONFIGURAITON` here

File: src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java [hunk 3]
```diff
@@ -124,6 +124,26 @@ public class RepositoryOptions extends OptionsBase {
...
+
+  @Option(
+      name = "http_connector_retry_max_timeout",
+      defaultValue = "0",
+      documentationCategory = OptionDocumentationCategory.BAZEL_CLIENT_OPTIONS,
+      effectTags = {OptionEffectTag.UNKNOWN},
+      help =
+          "The maximum timeout for http download retries. If set to 0, "
+              + "a default value is used.")
+  public int httpConnectorRetryMaxTimeout;
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java:145]: use the `Duration` type instead: see example https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/buildeventservice/BuildEventServiceOptions.java;drc=f63bb58023840454ba4997175487865b0ae8ec74;l=43

File: src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java (hunk 1 above)
Comment (exoson) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java]: Done. Made the other one match the current behavior of no maximum by default.

File: src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java (hunk 2 above)
Comment (exoson) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java]: Done

File: src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java (hunk 3 above)
Comment (exoson) [reply src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryOptions.java]: Done

**Review by Wyverald — APPROVED** [review]

**Pull request closed without merging**